hazelcast/src/test/java/com/hazelcast/nio/tcp/nonblocking/iobalancer/IOBalancerStressTest.java

```diff
@@ -74,7 +74,7 @@ public void testEachConnectionUseDifferentSelectorEventually() {
         IMap<Integer, Integer> map = instance1.getMap(randomMapName());
 
         long deadLine = System.currentTimeMillis() + TEST_DURATION_SECONDS * 1000;
-        for (int i = 0; System.currentTimeMillis() < deadLine; i++) {
+        for (int i = 0; System.currentTimeMillis() < deadLine && map.size() < 1000; i++) {
```

Comment:
This condition will terminate either when deadline passes or when map size is > 1000, thus it does not guarantee that 1000 put operations have been completed at the end of the loop and we may still face a condition under which deadline has passed (therefore for loop has terminated) but a small number of put operations has been completed. I think the time component in the condition should be eliminated if you want to make sure that 1000 operations are completed by the end of for loop.